addons/purchase_stock/tests/test_stockvaluation.py

```diff
@@ -934,18 +934,13 @@ def _now(*args, **kwargs):
         full_reconcile = stock_line.full_reconcile_id
         self.assertTrue(full_reconcile.exists())
 
-        reconciled_lines = full_reconcile.reconciled_line_ids - stock_line
-        self.assertEqual(len(reconciled_lines), 2)
+        reconciled_lines = full_reconcile.reconciled_line_ids
+        self.assertEqual(len(reconciled_lines), 3)
 
         stock_journal_line = reconciled_lines.filtered(lambda l: l.journal_id == self.stock_journal)
         self.assertEqual(stock_journal_line.amount_currency, -30.00)
         self.assertAlmostEqual(stock_journal_line.balance, -42.86)
 
-        exhange_diff_journal = company.currency_exchange_journal_id.exists()
-        exchange_stock_line = reconciled_lines.filtered(lambda l: l.journal_id == exhange_diff_journal)
-        self.assertEqual(exchange_stock_line.amount_currency, 0.00)
-        self.assertAlmostEqual(exchange_stock_line.balance, 27.86)
```

Comment:
why?

Reply:
the reconciliation is different when reconciling a line with a foreign currency to a line using the currency of the company